testing/fil_token_integration/actors/basic_transfer_actor/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+use cid::{multihash::Code, Cid};
+use fil_fungible_token::{
+    receiver::types::{FRC46TokenReceived, UniversalReceiverParams, FRC46_TOKEN_TYPE},
+    token::types::TransferParams,
+};
+use frc42_dispatch::{match_method, method_hash};
+use fvm_ipld_blockstore::Block;
+use fvm_ipld_encoding::tuple::{Deserialize_tuple, Serialize_tuple};
+use fvm_ipld_encoding::{de::DeserializeOwned, RawBytes, DAG_CBOR};
+use fvm_sdk as sdk;
+use fvm_shared::{address::Address, bigint::Zero, econ::TokenAmount, error::ExitCode};
+use sdk::NO_DATA_BLOCK_ID;
+
+/// Grab the incoming parameters and convert from RawBytes to deserialized struct
+pub fn deserialize_params<O: DeserializeOwned>(params: u32) -> O {
+    let params = sdk::message::params_raw(params).unwrap().1;
+    let params = RawBytes::new(params);
+    params.deserialize().unwrap()
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+struct TransferActorState {
+    operator_address: Option<Address>,
+    token_address: Option<Address>,
+}
+
+impl TransferActorState {
+    fn load(cid: &Cid) -> Self {
+        let data = sdk::ipld::get(cid).unwrap();
+        fvm_ipld_encoding::from_slice::<Self>(&data).unwrap()
+    }
+
+    fn save(&self) -> Cid {
+        let serialized = fvm_ipld_encoding::to_vec(self).unwrap();
+        let block = Block { codec: DAG_CBOR, data: serialized };
+        sdk::ipld::put(Code::Blake2b256.into(), 32, block.codec, block.data.as_ref()).unwrap()
+    }
+}
+
+#[no_mangle]
+fn invoke(input: u32) -> u32 {
```

Comment:
Could you describe the basic behaviour that this implements?

Reply:
Added a comment describing how the actor works